app/models/host.rb

```diff
@@ -86,6 +99,13 @@ class Host < ApplicationRecord
     left_outer_joins(:test_result_profiles)
   }
 
+  scope :joins_test_result_profiles_with_failed_rules, lambda {
+    # Needed to GROUP BY Host columns to work correctly with sortable_by :rules_failed
+    left_outer_joins(test_result_profiles: :rule_results).where(
+      rule_results: { result: RuleResult::FAILED }
+    ).group(HOST_COLUMNS)
```

Comment:
Grouping by the host columns was needed for `sortable_by :rules_failed` (like mentioned in the comment) but not sure if this is the best place the add it 🤔

Reply:
A grouping can mess up the object in activerecord in ways that can not be handled by the serializer sometimes. What you could do is to "hide" it in a subquery. However, I would need a bit more context of why the grouping here is needed.